sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -57,11 +66,15 @@ case class PartitionedFile(
 class FileScanRDD(
     @transient private val sparkSession: SparkSession,
     readFunction: (PartitionedFile) => Iterator[InternalRow],
-    @transient val filePartitions: Seq[FilePartition])
+    @transient val filePartitions: Seq[FilePartition],
+    val requiredSchema: StructType = StructType(Seq.empty),
```

Comment:
nit: `dataSchema` is probably a better name in this context. We need to join the data rows with metadata col w.r.t. this schema.

Reply:
Make sense, wondering if it will confuse with real `dataSchema` (e.g. in CSV, requiredSchema <= dataSchema)?

How about `rowSchema`, `rowDataSchema`, `dataRowSchema`, `readSchema`?

Reply:
Also why is the default value for `requiredSchema` passed as empty? I understand that this is used only when `metadataStructCol` is passed but this dependency is not very clear.

Reply:
`readSchema` sounds good. Can we always pass this parameter?

Reply:
sure!